Modules/xSharePoint/DSCResources/MSFT_xSPSearchTopology/MSFT_xSPSearchTopology.psm1

```diff
@@ -83,7 +82,14 @@ function Set-TargetResource
 
         # Ensure the search service instance is running on all servers
         foreach($searchServer in $AllSearchServers) {
-            $searchService = Get-SPEnterpriseSearchServiceInstance -Identity $searchServer
+            
+            $searchService = Get-SPEnterpriseSearchServiceInstance -Identity $searchServer -ErrorAction SilentlyContinue
+            if ($searchService -eq $null) {
+                $domain = (Get-CimInstance -ClassName Win32_ComputerSystem).Domain
+                $searchServer = "$searchServer.$domain"
+                $searchService = Get-SPEnterpriseSearchServiceInstance -Identity $searchServer    
+            }
+            
```

Comment:
What happens if after this change the result is still $null?? As far as I can see there is no additional check.